src/umessage.rs

```diff
@@ -64,10 +66,37 @@ impl From<&str> for UMessageError {
 }
 
 impl UMessage {
-    /// Extracts the payload-contained protobuf message from a `UMessage`.
+    // Deserialize a proto-generated `Message`-type from payload `Bytes`, according to `UPayloadFormat`
+    // Will only succeed if payload format is one of
+    // - `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF`
+    // - `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY`
+    pub(crate) fn deserialize_protobuf_bytes<T: Message + Default>(
```

Comment:
FMPOV we should move this out of the UMessage struct as it has meaning independently from it